File: day1.rb
def sol1()
    source = File.open("day1.txt")
    content = source.readlines.map(&:to_i)
    content.each{
        |x|
        content.each{
            |y|
                if ((x + y) == 2020)
                    puts x
                    puts y
                    puts x*y
                    return
                end
        }
    }
    source.close
    end

def sol2()
    source = File.open("day1.txt")
    content = source.readlines.map(&:to_i)
    content.each{
        |x|
        content.each{
            |y|
            content.each{
                |z|
                if ((x + y + z) == 2020)
                    puts x
                    puts y
                    puts z
                    puts x*y*z
                    return
                end
            }
        }
    }
    source.close
    end

sol1()
sol2()

File: day1.py
filename = "day1ex.txt"

def sol():
    with open(filename, 'r') as f:
        r = list(map(lambda x: int(x.strip("\n")),f.readlines()))
        for x in r:
            for y in r:
                if ((x+y) == 2020):
                    print(f"""{x*y}""")
                    return x*y

sol()

def sol2():
    with open(filename, 'r') as f:
        r = list(map(lambda x: int(x.strip("\n")),f.readlines()))
        for x,y,z in r,r,r:
            print(x,y,z)
            # for y in r.remove(x):
            #     for z in r.remove(x,y):
            #         if ((x+y+z) == 2020):
            #             print(f"""{x*y*z}""")
            #             return x*y*z

sol2()
